website/blog/four-approaches-for-documenting-apis.md

```diff
@@ -0,0 +1,126 @@
+
+### Cons
+
+**Hard to keep updated:** Design is an ongoing process; new changes should be made to the design first. However, if a hotfix or new requirement is urgent, the design process could be skipped, causing the documentation and code implementation to become out of sync. 
+
+**Chasing errors after implementation:** Changes on the design end can break implementations on the client side, causing developers to stop working on new features or other updates while finding and fixing the issue.
+
+## Method 3: Code First
+
+Code First, the traditional approach to API documentation, is a bottom-up approach that involves programmatically generating API docs from existing code. 
+
+### How It Works
+
+With the Code First approach, developers code the API directly from completed business requirements, leaving the creation of the documentation until later in the project. You can create documentation using the previously mentioned manual method or by generating machine-readable API definitions with [OpenAPI](https://www.openapis.org/) and frameworks such as [`express-openapi`](https://www.npmjs.com/package/express-openapi). 
+
+OpenAPI is an API description format for REST APIs that has become an industry standard for documenting APIs. With `express-openapi`, you can generate and validate OpenAPI documentation in JSON for Express apps. `express-openapi` can auto-generate documentation using the code definitions in your app or from the request and response schemas and parameters of your API spec. 
+
+### Pros 
+
+**More accurate:** Because your API is generated from existing code, the definition will always remain consistent with the API implementation. Consumers can be confident that the documentation is an accurate reflection of the code and that the API will work as expected. 
+
+**Faster delivery:** If your projects are small or need to be fast to market, the Code First approach might be a better option for you. Developers can get to work immediately, coding directly from requirements documents instead of after the design process, leaving documentation generation until the end of the project.
+
+### Cons 
+
+**Changes are hard to see:** Since changes occur in the code first and are then reflected in the generated API documents, there’s no way to compare previous versions of the API with the current version to validate those changes. 
+
+**Must call each endpoint to generate docs:** To generate docs, you still need to fill in the URI; method; parameters, if any; headers; and authentication details for the API. 
+
+## Method 4: Documentation Diffs with Optic
+
+[Optic](https://useoptic.com/) is a version control system and central repository for API contracts that function much like GitHub. It helps you document your APIs as you build them by observing development traffic to learn the behavior of your API. It presents the API changes it finds as diffs that can be reviewed, approved, or documented before your API is released. 
```

Comment:
It does only mention development traffic, so if we want to clarify further, we should.